processor/transformprocessor/internal/common/metrics.go

```diff
@@ -70,87 +70,74 @@ func (d dataPointStatements) Capabilities() consumer.Capabilities {
 }
 
 func (d dataPointStatements) ConsumeMetrics(ctx context.Context, md pmetric.Metrics) error {
-	for i := 0; i < md.ResourceMetrics().Len(); i++ {
-		rmetrics := md.ResourceMetrics().At(i)
-		for j := 0; j < rmetrics.ScopeMetrics().Len(); j++ {
-			smetrics := rmetrics.ScopeMetrics().At(j)
-			metrics := smetrics.Metrics()
-			for k := 0; k < metrics.Len(); k++ {
-				metric := metrics.At(k)
-				var err error
+	var errors error
+	md.ResourceMetrics().RemoveIf(func(rmetrics pmetric.ResourceMetrics) bool {
+		rmetrics.ScopeMetrics().RemoveIf(func(smetrics pmetric.ScopeMetrics) bool {
+			smetrics.Metrics().RemoveIf(func(metric pmetric.Metric) bool {
 				switch metric.Type() {
 				case pmetric.MetricTypeSum:
-					err = d.handleNumberDataPoints(ctx, metric.Sum().DataPoints(), metrics.At(k), metrics, smetrics.Scope(), rmetrics.Resource())
+					metric.Sum().DataPoints().RemoveIf(func(datapoint pmetric.NumberDataPoint) bool {
+						tCtx := ottldatapoint.NewTransformContext(datapoint, metric, smetrics.Metrics(), smetrics.Scope(), rmetrics.Resource())
+						remove, err := executeStatements(ctx, tCtx, d)
```

Comment:
This section is repeated for each case, could we move the common code outside of the switch? The switch would then only be used get the metric context `tCtx`.

Reply:
Moved datapoint handle back into their own funcs.  I left out `callFunction` still because reducing the internal code for each `RemoveIf` function didn't end up reducing complexity since each `handle*` function has to do multierr logic.